src/fmu/sumo/explorer/objects/_search_context.py

```diff
@@ -1502,6 +1502,16 @@ def __verify_aggregation_operation(
             "key"
         ]
         classname = sres["aggregations"]["class"]["buckets"][0]["key"]
+
+        for i in self._query["bool"]["must"]:
+            if (
+                (i.get("terms") and "fmu.realization.id" in i["terms"])
```

Comment:
Note: this functionality needs to be implemented for the `async` version of this function as well - you'll need to call `length_async()` instead of `len()` for that.